.github/workflows/acceptance.yml

```diff
@@ -0,0 +1,196 @@
+name: Acceptance
+
+# Optional, enabling concurrency limits: https://docs.github.com/en/actions/using-jobs/using-concurrency
+#concurrency:
+#  group: ${{ github.ref }}-${{ github.workflow }}
+#  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  actions: none
+  checks: none
+  contents: none
+  deployments: none
+  id-token: none
+  issues: none
+  discussions: none
+  packages: none
+  pages: none
+  pull-requests: none
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages
+      - metakitty
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'metsploit-framework.gemspec'
+      - 'Gemfile.lock'
+      - 'data/templates/**'
+      - 'modules/payloads/**'
+      - 'lib/msf/core/payload/**'
+      - 'lib/msf/core/**'
+      - 'spec/acceptance/**'
+      - 'spec/acceptance_spec_helper.rb'
+#   Example of running as a cron, to weed out flaky tests
+#  schedule:
+#    - cron: '*/15 * * * *'
+
+jobs:
+  # Run all test individually, note there is a separate final job for aggregating the test results
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-11
+          - windows-2019
+          - ubuntu-20.04
+        ruby:
+          - 3.0.2
+        meterpreter:
+          # Python
```

Comment:
Isn't Python2 still supported?

Reply:
It's no longer possible to run Python 2.7 natively on Github actions without Docker - https://github.com/actions/setup-python/issues/672 I can revisit this decision again in the future, but we won't get Windows/OSX Python 2.7 testing

For visibility - over in the metasploit-payloads repo I introduced a unit test suite for the Python Meterpreter which runs multiple Python versions as well https://github.com/rapid7/metasploit-payloads/pull/651 It'd be great to add something similar for the PHP payload, but I don't believe there's any native test suite runners that come bundled with PHP like Python's `unittest`

Reply:
For PHP, there is [PHPUnit]( https://phpunit.de/ ) (to test PHP code behaviour) and/or [PHP's native testsuite]( https://qa.php.net/running-tests.php ) (to test the interpreter behaviour).

Reply:
Just created a separate issue for adding a basic PHP Meterpreter test suite https://github.com/rapid7/metasploit-payloads/issues/668 :+1:

For this PR though, there won't be Python 2.7 integration tests running natively on osx/windows/linux due to it being dropped by Github